src/templates/main.tsx

```diff
@@ -41,11 +41,12 @@ export const getPath: GetPath<TemplateProps> = ({ document }) => {
     // temporary: guard for generated repo-based static page
     return `static-${Math.floor(Math.random() * (10000 - 1))}`;
   }
-  return document.slug
-    ? document.slug
-    : `${document.locale}/${document.address.region}/${document.address.city}/${
+  const localePath = document.locale !== "en" ? `${document.locale}/` : "";
+  return document.address
+    ? `${localePath}${document.address.region}/${document.address.city}/${
         document.address.line1
-      }-${document.id.toString()}`;
+      }-${document.id.toString()}`
+    : document.id.toString();
```

Comment:
You still want the localePath for non-location-like entities or else it's going to path conflict if there are multiple locales.